hydromt/models/model_utils.py

```diff
@@ -0,0 +1,144 @@
+
+    Returns
+    -------
+    Dict[str, xr.Dataset]
+        dict of xarray.Dataset
+    """
+    ncs = dict()
+    fns = glob.glob(join(root, fn))
+    if "chunks" not in kwargs:  # read lazy by default
+        kwargs.update(chunks="auto")
+    for fn in fns:
+        name = basename(fn).split(".")[0]
+        logger.debug(f"Reading model file {name}.")
+        # Load data to allow overwritting in r+ mode
+        if load:
+            ds = xr.open_dataset(fn, mask_and_scale=mask_and_scale, **kwargs).load()
+            ds.close()
+        else:
+            ds = xr.open_dataset(fn, mask_and_scale=mask_and_scale, **kwargs)
+        # set geo coord if present as coordinate of dataset
+        if GEO_MAP_COORD in ds.data_vars:
+            ds = ds.set_coords(GEO_MAP_COORD)
+        # single-variable Dataset to DataArray
+        if single_var_as_array and len(ds.data_vars) == 1:
+            (ds,) = ds.data_vars.values()
+        ncs.update({name: ds})
+    return ncs
+
+
+def write_nc(
+    nc_dict: XArrayDict,
+    fn: str,
+    root,
+    logger: Logger,
+    temp_data_dir=None,
+    gdal_compliant: bool = False,
+    rename_dims: bool = False,
+    force_sn: bool = False,
+    **kwargs,
+) -> DeferedFileClose | None:
+    """Write dictionnary of xarray.Dataset and/or xarray.DataArray to netcdf files.
+
+    Possibility to update the xarray objects attributes to get GDAL compliant NetCDF
+    files, using :py:meth:`~hydromt.raster.gdal_compliant`.
+    The function will first try to directly write to file. In case of
+    PermissionError, it will first write a temporary file and add to the
+    self._defered_file_closes attribute. Renaming and closing of netcdf filehandles
+    will be done by calling the self._cleanup function.
+
+    key-word arguments are passed to :py:meth:`xarray.Dataset.to_netcdf`
+
+    Parameters
+    ----------
+    nc_dict: dict
+        Dictionary of xarray.Dataset and/or xarray.DataArray to write
+    fn: str
+        filename relative to model root and should contain a {name} placeholder
+    gdal_compliant: bool, optional
+        If True, convert xarray.Dataset and/or xarray.DataArray to gdal compliant
+        format using :py:meth:`~hydromt.raster.gdal_compliant`
+    rename_dims: bool, optional
+        If True, rename x_dim and y_dim to standard names depending on the CRS
+        (x/y for projected and lat/lon for geographic). Only used if
+        ``gdal_compliant`` is set to True. By default, False.
+    force_sn: bool, optional
+        If True, forces the dataset to have South -> North orientation. Only used
+        if ``gdal_compliant`` is set to True. By default, False.
+    **kwargs:
+        Additional keyword arguments that are passed to the `to_netcdf`
+        function.
+    """
+    for name, ds in nc_dict.items():
+        if not isinstance(ds, (xr.Dataset, xr.DataArray)) or len(ds) == 0:
+            logger.error(f"{name} object of type {type(ds).__name__} not recognized")
```

Comment:
should this be `self.logger`? currently you're always using the logger defined at the top instead of one the user provides